src/base/utils/misc.cpp

```diff
@@ -249,49 +284,96 @@ void Utils::Misc::shutdownComputer(const ShutdownDialogAction &action)
+        {
+            const auto &unitString = decimalPrefixBitsUnits[static_cast<int>(unit)];
+            ret = QCoreApplication::translate("misc", unitString.source, unitString.comment);
+            break;
+        }
+    case unitType::decimalPrefixBytes:
+        {
+            const auto &unitString = decimalPrefixBytesUnits[static_cast<int>(unit)];
+            ret = QCoreApplication::translate("misc", unitString.source, unitString.comment);
+            break;
+        }
+     case unitType::binaryPrefixBits:
+        {
+            const auto &unitString = binaryPrefixBitsUnits[static_cast<int>(unit)];
+            ret = QCoreApplication::translate("misc", unitString.source, unitString.comment);
+            break;
+        }
+     case unitType::binaryPrefixBytes:
+        {
+            const auto &unitString = binaryPrefixBytesUnits[static_cast<int>(unit)];
+            ret = QCoreApplication::translate("misc", unitString.source, unitString.comment);
+        }
+    }
+
     if (isSpeed)
         ret += QCoreApplication::translate("misc", "/s", "per second");
+
     return ret;
 }
 
-QString Utils::Misc::friendlyUnit(const qint64 bytes, const bool isSpeed)
+QString Utils::Misc::friendlyUnit(qint64 bytes, const bool isSpeed)
 {
-    const std::optional<SplitToFriendlyUnitResult> result = splitToFriendlyUnit(bytes);
+    unitType prefix = isSpeed ? Preferences::instance()->transferSpeedUnitsPrefix() :
+                                Preferences::instance()->fileSizeUnitsPrefix();
+
+    if (prefixIsInBits(prefix))
+        bytes *= 8;  // User wants bits
+
+    const std::optional<SplitToFriendlyUnitResult> result = splitToFriendlyUnit(bytes, prefix);
     if (!result)
         return QCoreApplication::translate("misc", "Unknown", "Unknown (size)");
+
     return Utils::String::fromDouble(result->value, friendlyUnitPrecision(result->unit))
            + QString::fromUtf8(C_NON_BREAKING_SPACE)
-           + unitString(result->unit, isSpeed);
+           + unitString(result->unit, prefix, isSpeed);
 }
 
 int Utils::Misc::friendlyUnitPrecision(const SizeUnit unit)
 {
     // friendlyUnit's number of digits after the decimal point
     switch (unit)
     {
-    case SizeUnit::Byte:
+    case SizeUnit::B:
         return 0;
-    case SizeUnit::KibiByte:
-    case SizeUnit::MebiByte:
+    case SizeUnit::Ki:
+    case SizeUnit::Me:
         return 1;
-    case SizeUnit::GibiByte:
+    case SizeUnit::Gi:
         return 2;
     default:
         return 3;
     }
 }
 
-qlonglong Utils::Misc::sizeInBytes(qreal size, const Utils::Misc::SizeUnit unit)
+qlonglong Utils::Misc::sizeInBytes(qreal size, const Utils::Misc::SizeUnit unit, Utils::Misc::unitType prefix)
 {
+    int multiplier = prefixIsDecimal(prefix) ? 1000 : 1024;
+
     for (int i = 0; i < static_cast<int>(unit); ++i)
-        size *= 1024;
+        size *= multiplier;
     return size;
 }
 
+bool Utils::Misc::prefixIsInBits(const unitType prefix)
```

Comment:
Why `prefix`? Seems it is about `unit`.

Reply:
Yes, you're right. I copied the function name below (prefixIsDecimal) and the result is not good. I think a better name would be "unitTypeIsInBits"